cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -141,35 +141,45 @@ func createPod(ctx context.Context, events publisher, req *task.CreateTaskReques
 		id:     req.ID,
 		host:   parent,
 	}
-	// TOOD: JTERRY75 - There is a bug in the compartment activation for Windows
+
+	var netSetup hcsoci.NetworkSetup
+	if parent != nil {
+		if parent.NCProxyEnabled() {
+			caAddr := fmt.Sprintf(podComputeAgentAddrFmt, req.ID)
+			s, ttrpcListener, err := setupComputeAgent(caAddr, parent)
+			if err != nil {
+				return nil, fmt.Errorf("failed to setup compute agent service: %s", err)
+			}
+			serveComputeAgent(ctx, s, ttrpcListener)
+			netSetup = &hcsoci.ExternalNetworkSetup{
+				VM:          parent,
+				CaAddr:      caAddr,
+				ContainerID: req.ID,
+			}
+		} else {
+			netSetup = &hcsoci.LocalNetworkSetup{VM: parent}
+		}
+	}
+
+	// TODO: JTERRY75 - There is a bug in the compartment activation for Windows
 	// Process isolated that requires us to create the real pause container to
 	// hold the network compartment open. This is not required for Windows
 	// Hypervisor isolated. When we have a build that supports this for Windows
 	// Process isolated make sure to move back to this model.
+
+	// For WCOW we fake out the init task since we dont need it. We only
+	// need to provision the guest network namespace if this is hypervisor
+	// isolated. Process isolated WCOW gets the namespace endpoints
+	// automatically.
+	nsid := ""
 	if isWCOW && parent != nil {
```

Comment:
In the lcow/workload wcow route, we call into `newHcsTask` which eventually calls into `CreateContainer` which will setup the networking. Since we don't actually have a pause container for wcow pod, we call into `newWcowPodSandboxTask` which doesn't do much. I wonder if it would make sense to refactor this network setup code and move that to `newWcowPodSandboxTask`? This would just make it a little more consistent in terms of where to find what code for each path.

Reply:
Yea this would clean up even the non ncproxy logic. I don't see anything wrong with this off the top of my head